internal/controllers/synthetic_devices_controller.go

```diff
@@ -261,6 +275,11 @@ func (vc *SyntheticDevicesController) MintSyntheticDevice(c *fiber.Ctx) error {
 		return fiber.NewError(fiber.StatusInternalServerError, "synthetic device minting request failed")
 	}
 
+	if err := vc.handleDeviceAPIIntegrationCreation(c.Context(), tx, req, vNFT.UserDeviceID.String, integration); err != nil {
```

Comment:
This is a heavy signature. The controller is getting heavy too! Let's think about this.

I think one divide is to try to keep the web2 and 3 stuff separate.